lib/compile/index.js

```diff
@@ -0,0 +1,96 @@
+import most from "most";
+import escodegen from "escodegen";
+
+import applyPlugins from "./apply-plugins";
+import construct from "./construct";
+import compileModules from "./modules/compile";
+import bootstrapBundles from "./bundles/bootstrap";
+import hashBundle from "./bundles/hash";
+import interpolateFilename from "./bundles/interpolate-filename";
+import dedupeExplicit from "./bundles/dedupe-explicit";
+import dedupeImplicit from "./bundles/dedupe-implicit";
+
+export default function compile (opts, compilation) {
+  compilation = compilation || {
+    cache: {
+      modulesByAbsPath: Object.create(null)
+    },
+    applyPlugins: (ev, ...args) => applyPlugins(opts.plugins, ev, ...args),
```

Comment:
Oh I see.  No, it won't.  `ev` represents particular injection points in the compilation process, allowing plugins to modify data at that point.  However, as I said, I'd like to completely re-evaluate this.

A singleton accessible everywhere would be so much simpler, but in the back of my mind I'm thinking about supporting two concurrent compilations in the same process.  So trying to keep this as purely functional and side-effect free as possible.